internal/quickfort/orders.lua

```diff
@@ -115,18 +114,20 @@ local function get_num_items(b)
 end
 
 function create_orders(ctx)
-    for k,order_spec in pairs(ctx.order_specs) do
+    for k,order_spec in pairs(ctx.order_specs or {}) do
         local quantity = math.ceil(order_spec.quantity)
         log('ordering %d %s', quantity, k)
         stockflow.create_orders(order_spec.order, quantity)
         table.insert(ctx.stats, {label=k, value=quantity})
     end
 end
 
+-- ensure we don't reinit this; it contains allocated memory
```

Comment:
You'll want to drop the `local` keyword to avoid reallocating memory - local variables don't persist across script runs, so `reactions` will always be nil before this line.

Granted, under normal circumstances, this file should only be "run" once by end-users, when the main quickfort entrypoint imports it. (Other things that would cause this to be re-run include editing this file or running `internal/quickfort/orders` from the console.) I'm pretty sure running just `quickfort` multiple times would only cause collect_reactions to be called once.

Reply:
I just realized that stockflow is probably regenerating this list on world load because there could be different reactions available in each world. A lot of the furniture reactions are hardcoded, though, but custom workshops could vary - do you support those?

This should be fine for the majority of furniture reactions, in any case. I'd be fine with refactoring stockflow.lua a bit later to support this use case better.

Reply:
The only orders quickfort would ever generate will be for stuff that is buildable. can custom workshops produce buildable items? Is the list of what can be placed with the `b`uild menu variable?

Reply:
I was talking about building actual custom workshops... which I just realized you can't build through the manager. (They affect the `b-W` menu, and I saw support for building workshops in one of the library blueprints earlier, which is why I thought of custom workshops, but the presence of custom workshops alone won't affect what orders are available. So this is fine - I was confusing some unrelated things.

Reply:
done